### PR TITLE
Added drag drop feature for image

### PR DESCRIPTION
**Pull request by lazyXghost**

### CODE REVIEW

File: css/style.css [hunk 1]
```diff
@@ -25,6 +25,20 @@
 	padding: 10px;
 	margin-top: 0px;
 }
+.dragDropDisplay {
+  background: linear-gradient(to left, rgba(0, 0, 0, 0.9), rgba(0, 0, 0, 0.2)), url('/images/grass_block.png') top center repeat-x;
```
Comment (csemrau) [inline css/style.css:29]: Is that image a copy of the Minecraft grass block texture or some other potentially copyrighted assets of the game? 🚨

Comment (lazyXghost) [reply css/style.css:29]: This image is only a placeholder image, anyways I am going to change it with a better image, and will keep the copyrighted content in mind

File: scripts/main.js [hunk 2]
```diff
@@ -6,6 +6,7 @@ Minecraft Pixel Art Maker
 /** Colours that are selected on creation of a new image form. */
 const default_palette = Array.from(Colours.keys()).join(' ');
 
+let counter = 0;
```
Comment (gd-codes) [inline scripts/main.js:9]: This is probably better wrapped in a function along with the drag/drop event listener callbacks
- won't add to the global namespace
- will be separate for each image form instance

File: scripts/main.js [hunk 3]
```diff
@@ -216,6 +258,25 @@ function newImageUpload(uid, {fnName = "", active = true} = {}) {
...
 
+/** 
+ * Grab content of dropped images and passes the files to file input handler.
+ * @param {Event} event
+ * @param {UID} uid - Uid for the imp input field
+ */
+function imgDropHandler(event, uid) {
+  event.preventDefault();
+  [...event.dataTransfer.items].forEach((item, i) => {
+    if (item.kind === "file") {
```
Comment (gd-codes) [inline scripts/main.js:269]: Please add a check for the mimetype and only allow image files

File: css/style.css [hunk 4]
```diff
@@ -25,6 +25,20 @@
...
+  z-index: 1000;
+  font-family: Minecraft, Verdana, sans-serif;
+}
+.dragDropButton {
```
Comment (gd-codes) [inline css/style.css:33]: Allow cursor movement to pass through to the dragDropDisplay for catching events.
```suggestion
.dragDropButton {
  pointer-events: none;
```

**Review by gd-codes — CHANGES_REQUESTED** [review]

File: images/dragover_bg.png
Comment (gd-codes) [inline images/dragover_bg.png:1]: Replaced the previous images with a simple dirt texture. Can still consider a different background later on

File: scripts/main.js [hunk 5]
```diff
@@ -214,8 +214,75 @@ function newImageUpload(uid, {fnName = "", active = true} = {}) {
...
+  handleDragDropEvents(uid);
 }
 
+/**
+ * Respond to file drag/drop attempts over an image form.
+ * @param {String} uid - Identifies the image form
+ */
+function handleDragDropEvents(uid) {
+  // Counter helps manage repeatedly fired drag enter+leave events
+  var counter = 0;
```
Comment (gd-codes) [inline scripts/main.js:226]: Encapsulated the callback attachments in a function with the `counter` var to remove it from global namespace.

**Pull request merged**